pkg/tfbridge/info.go

```diff
@@ -329,9 +329,17 @@ type ResourceInfo struct {
 	// outputs before they are stored. In particular, it can be used as a last resort hook to
 	// make corrections in the default translation of the resource state from TF to Pulumi.
 	// Should be used sparingly.
-	TransformOutputs func(ctx context.Context, outputs resource.PropertyMap) (resource.PropertyMap, error)
+	TransformOutputs PropertyTransform
+
+	// Check, Diff, Read, Update and Delete refer to old inputs sourced from the
+	// Pulumi statefile. TransformFromState lets providers edit these outputs before they
+	// are accessed by other provider functions or by terraform. In particular, it can
+	// be used to perform upgrades on old pulumi state.  Should be used sparingly.
+	TransformFromState PropertyTransform
```

Comment:
This pairs well with TransformOutputs as the two together work toward allowing a provider to biject between state representations that the engine sees and saves into state files and the ones that the provider sees and feeds to TF.